src/utils/debugdraw_gl.cc

```diff
@@ -0,0 +1,455 @@
+        offset += sizeof(float) * 3;
+
+        glEnableVertexAttribArray(1); // in_ColorPointSize (vec4)
+        glVertexAttribPointer(
+            /* index     = */ 1,
+            /* size      = */ 4,
+            /* type      = */ GL_FLOAT,
+            /* normalize = */ GL_FALSE,
+            /* stride    = */ sizeof(dd::DrawVertex),
+            /* offset    = */ reinterpret_cast<void *>(offset));
+
+        checkGLError(__FILE__, __LINE__);
+
+        // VAOs can be a pain in the neck if left enabled...
+        glBindVertexArray(0);
+        glBindBuffer(GL_ARRAY_BUFFER, 0);
+    }
+
+    //
+    // Text rendering vertex buffer:
+    //
+    {
+        glGenVertexArrays(1, &textVAO);
+        glGenBuffers(1, &textVBO);
+        checkGLError(__FILE__, __LINE__);
+
+        glBindVertexArray(textVAO);
+        glBindBuffer(GL_ARRAY_BUFFER, textVBO);
+
+        // NOTE: A more optimized implementation might consider combining
+        // both the lines/points and text buffers to save some memory!
+        glBufferData(GL_ARRAY_BUFFER, DEBUG_DRAW_VERTEX_BUFFER_SIZE * sizeof(dd::DrawVertex), nullptr, GL_STREAM_DRAW);
+        checkGLError(__FILE__, __LINE__);
+
+        // Set the vertex format expected by the 2D text:
+        std::size_t offset = 0;
+
+        glEnableVertexAttribArray(0); // in_Position (vec2)
+        glVertexAttribPointer(
+            /* index     = */ 0,
+            /* size      = */ 2,
+            /* type      = */ GL_FLOAT,
+            /* normalize = */ GL_FALSE,
+            /* stride    = */ sizeof(dd::DrawVertex),
+            /* offset    = */ reinterpret_cast<void *>(offset));
+        offset += sizeof(float) * 2;
+
+        glEnableVertexAttribArray(1); // in_TexCoords (vec2)
+        glVertexAttribPointer(
+            /* index     = */ 1,
+            /* size      = */ 2,
+            /* type      = */ GL_FLOAT,
+            /* normalize = */ GL_FALSE,
+            /* stride    = */ sizeof(dd::DrawVertex),
+            /* offset    = */ reinterpret_cast<void *>(offset));
+        offset += sizeof(float) * 2;
+
+        glEnableVertexAttribArray(2); // in_Color (vec4)
+        glVertexAttribPointer(
+            /* index     = */ 2,
+            /* size      = */ 4,
+            /* type      = */ GL_FLOAT,
+            /* normalize = */ GL_FALSE,
+            /* stride    = */ sizeof(dd::DrawVertex),
+            /* offset    = */ reinterpret_cast<void *>(offset));
+
+        checkGLError(__FILE__, __LINE__);
+
+        // Ditto.
+        glBindVertexArray(0);
+        glBindBuffer(GL_ARRAY_BUFFER, 0);
+    }
+}
+
+// ========================================================
+// Minimal shaders we need for the debug primitives:
+// ========================================================
+
+const char * DDRenderInterfaceCoreGL::linePointVertShaderSrc = R"(
+    #version 150
+
+    in vec3 in_Position;
+    in vec4 in_ColorPointSize;
+
+    out vec4 v_Color;
+    uniform mat4 u_MvpMatrix;
```

Comment:
Tweak these shaders to take the same per-camera uniform block everything else does, and a bunch of the other noise in this file goes away. (along with needing more copies of your view*proj matrix in main)